OpenRA.Mods.Common/Util.cs

```diff
@@ -155,5 +158,70 @@ public static IEnumerable<CPos> RandomWalk(CPos p, MersenneTwister r)
 				yield return p;
 			}
 		}
+
+		public static string FriendlyTypeName(Type t)
+		{
+			if (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(HashSet<>))
+				return "Set of {0}".F(t.GetGenericArguments().Select(FriendlyTypeName).ToArray());
+
+			if (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(Dictionary<,>))
+				return "Dictionary<{0},{1}>".F(t.GetGenericArguments().Select(FriendlyTypeName).ToArray());
```

Comment:
If you're going to use friendly names (which I think are silly) then this and Cached need to match.